Mathlib/Topology/Compactification/OnePointEquiv.lean

```diff
@@ -0,0 +1,166 @@
+/-
+Copyright (c) 2024 Bjørn Kjos-Hanssen. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bjørn Kjos-Hanssen
+-/
+import Mathlib.Data.Real.Basic
+import Mathlib.LinearAlgebra.Projectivization.Basic
+import Mathlib.Topology.Compactification.OnePoint
+
+/-!
+# OnePoint Compactification and Projectivization
+
+We construct a set-theoretic equivalence between
+`OnePoint K` and the projectivization `ℙ K (Fin 2 → K)`
+for an arbitrary field `K`.
+
+(This equivalence can be extended to a homeomorphism
+in the case `K = ℝ`, where `OnePoint ℝ` gets the topology of one-point compactification.
+This result is to be added in a different file.)
+
+## Main definitions
+
+* `divOnePoint`: Division on `K` taking values in `OnePoint K`.
+* `divSlope`: Lifting `divOnePoint` to a map `ℙ K (Fin 2 → K) → OnePoint K`.
+
+## Main results
+
+* `ℙ K (Fin 2 → K) ≃ OnePoint K`
+
+## Tags
+
+one-point extension, projectivization
+-/
+
+
+open scoped LinearAlgebra.Projectivization OnePoint
+open Projectivization Classical
+
+/-!
+### Definition and basic properties
+
+In this section we define `divOnePoint` and prove that it lifts to `divSlope`.
+-/
+
+/-- A modified division from a `DivisionRing` to its `OnePoint` extension,
+so as to satisfy `1 ÷ 0 = ∞`. -/
+noncomputable def divOnePoint {K : Type} [DivisionRing K] (a : K) (r : K): OnePoint K :=
+  ite (r ≠ 0) (a / r) ∞
+
+/-- Uncurried version of `divOnePoint`, with nonzeroness assumption. -/
+noncomputable def divOnePoint' {K : Type} [DivisionRing K]
+  (u : {v : Fin 2 → K // v ≠ 0}) : OnePoint K :=
```

Comment:
Normally, I would speak about `{v : Fin 2 → K} (hv : v ≠ 0)`. Any reason you are not doing this here?

Reply:
The same applies to most lemmas below. I checked a few of them; my impression is that the proof can always be fixed. Can you try to adapt your lemmas and proofs - feel free to ask on zulip in case you need help fixing the proofs!

Reply:
Thanks @grunweg for all the comments!
In my mind I was following the [projectivization](https://leanprover-community.github.io/mathlib4_docs/Mathlib/LinearAlgebra/Projectivization/Basic.html#projectivizationSetoid) definitions.
Also, compare
`(a b : {v : Fin 2 → K // v ≠ 0})`
versus
`(a b : Fin 2 → K) (ha : a ≠ 0) (hb : b ≠ 0)`.
In the first line the reader doesn't have to visually verify that the assumptions on `a` and `b` are indeed the same. It's basically the "Don't repeat yourself" (DRY) software principle.
What do you think?

(For just one variable I am fine with changing it from
`(a : {v : Fin 2 → K // v ≠ 0})`
to
`(a : Fin 2 → K) (ha : a ≠ 0)`; it seems both have their advantages.)

Reply:
About the DRY thing: I understand your point; my personal preference is more towards `{a b : Fin 2 → K} (ha : a ≠ 0) (hb : b ≠ 0)`, the arguments `a` and `b` being implicit.
One detail to take into account could be how this lemma will be used: is it used with `v` or with `a` and `ha`?

Reply:
OK I can go either way... In what I'm doing the lemma will be used with `v` in the sense that we are carrying around nonzero vectors, rather than carrying around vectors that may or may not be nonzero. Not sure if that answers your question.